### ISSUE TITLE
Upgrading owncloud 9.1.5-1.1 to 10.0.0-1.1 (debian_8.0) fails

### ISSUE THREAD

**Issue by jaharkes** [owncloud/core#27753]
### Steps to reproduce
1. Have installed owncloud + owncloud-deps-php5 from the owncloud 'stable-channel'
2. Run 'apt upgrade' to upgrade to the new stable version
3. Run 'sudo -u www-data php /var/www/owncloud/occ upgrade'

### Expected behaviour
Having an upgraded and working installation

### Actual behaviour
```
# sudo -u www-data php /var/www/owncloud/occ upgrade
ownCloud or one of the apps require upgrade - only a limited number of commands are available
You may use your browser or the occ upgrade command to do the upgrade
Set log level to debug
Updating database schema
Updated database
Drop old database tables

 Done
 28/28 [============================] 100%
Remove old (< 9.0) calendar/contact shares
 Done
 4/4 [============================] 100%
Fix permissions so avatars can be stored again
 Done
 2/2 [============================] 100%
Move user avatars outside the homes to the new location
 Starting ...
  3/13 [======>---------------------]  23%PHP Fatal error:  Class 'OCA\FederatedFileSharing\DiscoveryManager' not found in /var/www/owncloud/apps/files_sharing/lib/AppInfo/Application.php on line 84
```

### Server configuration
**Operating system**:
Debian 8.0

**Web server:**
Apache2 2.4.10-10+deb8u8

**Database:**
mariadb-server 10.0.30-0+deb8u1

**PHP version:**
php5 5.6.30+dfsg-0+deb8u1

**ownCloud version:** (see ownCloud admin page)
owncloud 10.0.0-1.1

**Updated from an older ownCloud or fresh install:**
updated from owncloud 9.1.5-1.1

**Where did you install ownCloud from:**
` http://download.owncloud.org/download/repositories/stable/Debian_8.0/`

**Signing status (ownCloud 9.0 and above):**

```
Login as admin user into your ownCloud and access 
http://example.com/index.php/settings/integrity/failed 
paste the results into https://gist.github.com/ and puth the link here.
```
Unable to do so because upgrade fails.

**The content of config/config.php:**

```
.{
    "system": {
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "datadirectory": "\/var\/lib\/owncloud\/data",
        "overwriteprotocol": "https",
        "overwritewebroot": "\/owncloud",
        "dbtype": "mysql",
        "version": "9.1.5.2",
        "appstoreenabled": true,
        "apps_paths": [
            {
                "path": "\/var\/www\/owncloud\/apps",
                "url": "\/apps",
                "writable": true
            }
        ],
        "dbname": "owncloud",
        "dbhost": "localhost",
        "dbtableprefix": "oc_",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "loglevel": 0,
        "asset-pipeline.enabled": false,
        "forcessl": true,
        "maintenance": true,
        "theme": "",
        "trashbin_retention_obligation": "auto",
        "memcache.local": "\\OC\\Memcache\\APCu",
        "updatechecker": false
    }
}
```

**List of activated apps:**

```
Enabled:
  - activity: 2.3.3
  - comments: 0.3.0
  - configreport: 0.1.1
  - dashboard: 2.0
  - dav: 0.2.8
  - external: 1.2
  - federatedfilesharing: 0.3.0
  - federation: 0.1.0
  - files: 1.5.1
  - files_external: 0.7.0
  - files_pdfviewer: 0.8.1
  - files_sharing: 0.10.0
  - files_texteditor: 2.1
  - files_trashbin: 0.9.0
  - files_versions: 1.3.0
  - files_videoplayer: 0.9.8
  - firstrunwizard: 1.1
  - gallery: 15.0.0
  - market: 0.1.0
  - notifications: 0.3.0
  - provisioning_api: 0.5.0
  - systemtags: 0.3.0
  - templateeditor: 0.1
  - updatenotification: 0.2.1
```

**Are you using external storage, if yes which one:** local/smb/sftp/...

local

**Are you using encryption:** yes/no

no

**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...

### Client configuration
**Browser:**

**Operating system:**

#### Web server error log
```
Insert your webserver log here
```

#### ownCloud log (data/owncloud.log)
```
{"reqId":"rFfqYPRe1w6UzINKSIDF","level":0,"time":"2017-04-27T10:58:56+00:00","remoteAddr":"","user":"--","app":"core","method":"--","url":"\/owncloud\/occ","message":"starting upgrade from 9.1.5.2 to 10.0.0.12"}
{"reqId":"rFfqYPRe1w6UzINKSIDF","level":3,"time":"2017-04-27T10:58:56+00:00","remoteAddr":"","user":"--","app":"core","method":"--","url":"\/owncloud\/occ","message":"Exception: {\"Exception\":\"OC\\\\RepairException\",\"Message\":\"Upgrade is not possible\",\"Code\":0,\"Trace\":\"#0 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/Repair.php(89): OC\\\\Repair\\\\Apps->run(Object(OC\\\\Repair))\\n#1 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/Updater.php(212): OC\\\\Repair->run()\\n#2 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/Updater.php(108): OC\\\\Updater->doUpgrade('10.0.0.12', '9.1.5.2')\\n#3 \\\/var\\\/www\\\/owncloud\\\/core\\\/Command\\\/Upgrade.php(249): OC\\\\Updater->upgrade()\\n#4 \\\/var\\\/www\\\/owncloud\\\/lib\\\/composer\\\/symfony\\\/console\\\/Command\\\/Command.php(262): OC\\\\Core\\\\Command\\\\Upgrade->execute(Object(Symfony\\\\Component\\\\Console\\\\Input\\\\ArgvInput), Object(Symfony\\\\Component\\\\Console\\\\Output\\\\ConsoleOutput))\\n#5 \\\/var\\\/www\\\/owncloud\\\/lib\\\/composer\\\/symfony\\\/console\\\/Application.php(826): Symfony\\\\Component\\\\Console\\\\Command\\\\Command->run(Object(Symfony\\\\Component\\\\Console\\\\Input\\\\ArgvInput), Object(Symfony\\\\Component\\\\Console\\\\Output\\\\ConsoleOutput))\\n#6 \\\/var\\\/www\\\/owncloud\\\/lib\\\/composer\\\/symfony\\\/console\\\/Application.php(189): Symfony\\\\Component\\\\Console\\\\Application->doRunCommand(Object(OC\\\\Core\\\\Command\\\\Upgrade), Object(Symfony\\\\Component\\\\Console\\\\Input\\\\ArgvInput), Object(Symfony\\\\Component\\\\Console\\\\Output\\\\ConsoleOutput))\\n#7 \\\/var\\\/www\\\/owncloud\\\/lib\\\/composer\\\/symfony\\\/console\\\/Application.php(120): Symfony\\\\Component\\\\Console\\\\Application->doRun(Object(Symfony\\\\Component\\\\Console\\\\Input\\\\ArgvInput), Object(Symfony\\\\Component\\\\Console\\\\Output\\\\ConsoleOutput))\\n#8 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/Console\\\/Application.php(160): Symfony\\\\Component\\\\Console\\\\Application->run(NULL, NULL)\\n#9 \\\/var\\\/www\\\/owncloud\\\/console.php(99): OC\\\\Console\\\\Application->run()\\n#10 \\\/var\\\/www\\\/owncloud\\\/occ(11): require_once('\\\/var\\\/www\\\/ownclo...')\\n#11 {main}\",\"File\":\"\\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/Repair\\\/Apps.php\",\"Line\":131}"}
{"reqId":"SHd0WkUFXjLNoAHSy2lG","level":0,"time":"2017-04-27T10:59:24+00:00","remoteAddr":"","user":"--","app":"core","method":"--","url":"\/owncloud\/occ","message":"starting upgrade from 9.1.5.2 to 10.0.0.12"}
{"reqId":"SHd0WkUFXjLNoAHSy2lG","level":1,"time":"2017-04-27T10:59:25+00:00","remoteAddr":"","user":"--","app":"core","method":"--","url":"\/owncloud\/occ","message":"Insert new users ..."}
{"reqId":"SHd0WkUFXjLNoAHSy2lG","level":1,"time":"2017-04-27T10:59:26+00:00","remoteAddr":"","user":"--","app":"core","method":"--","url":"\/owncloud\/occ","message":"Enable external storage: yes"}
{"reqId":"SHd0WkUFXjLNoAHSy2lG","level":3,"time":"2017-04-27T10:59:30+00:00","remoteAddr":"","user":"--","app":"PHP","method":"--","url":"\/owncloud\/occ","message":"Class 'OCA\\FederatedFileSharing\\DiscoveryManager' not found at \/var\/www\/owncloud\/apps\/files_sharing\/lib\/AppInfo\/Application.php#84"}
```

#### Browser log
```
Insert your browser log here, this could for example include:

a) The javascript console log
b) The network log 
c) ...
```

**Comment by kdslkdsaldsal** [comment]
@jaharkes This is a duplicate of https://github.com/owncloud/core/issues/26297

**Comment by jaharkes** [comment]
I figured this was a different problem because that described an upgrade 7 months ago from 9.0.3 to 9.1.1 on CentOS, while I was already at 9.1.5.

Looking at the suggested problem causes and solutions.

- My installation is not using federated sharing or LDAP.
- Running `occ maintenance:repair` completed successfully, but didn't help the upgrade.

- I managed to get the upgrade to complete after renaming `apps/federatedfileshareing/lib/DiscoveryManager.php` as described in https://github.com/owncloud/core/issues/26297#issuecomment-276384679. And patching `lib/autoloader.php` as described in https://github.com/owncloud/core/issues/26297#issuecomment-279751449.

**Comment by kdslkdsaldsal** [comment]
@jaharkes So why keep this open then? :-) It doesn't help if multiple issues about the very same problem are open.

**Comment by jaharkes** [comment]
Because I don't know if any developers are aware that this, at least 7 month old, problem still exists in the current code base. They can mark this a duplicate and close it if they want.

This happens with a release that was pushed out last night to the current Debian stable channel, and so other Debian users that hit the same problem may come around in the near term to report the exact same issue. Leaving this report open for a day or so should catch a bunch of "me too's"  and give them the actual steps that fixed the problem in my case.

Aside from avoid additional duplicate bug reports, now others do not have to read through a long chain of comments which contains various solutions that I tried and determined do not actually fix the problem. The working solution I found was pieced together from a combination of two separate comment that I linked to in my previous post and it really just allows the failed upgrade to complete. It doesn't actually fix the actual problem why this DiscoveryManager class is not found during the upgrade process.

**Comment by kdslkdsaldsal** [comment]
While some of your point might be valid its IMHO still recommended to collect all that info in https://github.com/owncloud/core/issues/26297. Only a few people are affected by this and it helps way more to somehow get similarities between your installations and collect them at one point instead of spreading those at the mentioned multiple issues.

**Comment by jaharkes** [comment]
Actually that particular comment is trying to debug some access denied and CSRF errors in the web interface that do not seem to be (directly) related to the original bug report about a class loading failure during upgrade.

Maybe those errors are fallout from the failed upgrade, but in my case I managed to allow the upgrade to complete and am not seeing such access errors with any web clients here.

I proposed a possible reason (https://github.com/owncloud/core/issues/26297#issuecomment-297812741)  for why the class cannot be discovered during upgrade when the filename is CamelCased, but succeeds when it is lowercased. Although I have no experience working with PHP it seems rather obvious from the code that the autoloader only tries to load lower cased filenames, which might work on WIndows but fails on any system with a case-sensitive file system.
